src/OmniSharp.MSBuild/ProjectSystem.cs

```diff
@@ -209,5 +210,10 @@ async Task<object> IProjectSystem.GetProjectModelAsync(string filePath)
 
             return new MSBuildProjectInfo(projectFileInfo);
         }
+
+        public async Task HasCompletedUpdateRequest()
```

Comment:
I think this needs a better name. "HasCompletedUpdateRequest" implies that this would return a boolean and doesn't really indicate that it's asynchronous.

How about `WaitForInitializationToCompletionAsync`?

Reply:
Or, if you decide to introduce a different interface for this behavior, maybe `HasPendingUpdatesAsync` that returns a `Task<bool>`?